classes/scheduledTask/ScheduledTask.inc.php

```diff
@@ -81,17 +96,26 @@ function getName() {
 	 * SCHEDULED_TASK_MESSAGE_TYPE... constants.
 	 */
 	function addExecutionLogEntry($message, $type = null) {
-		$log = $this->_executionLog;
+		$logFile = $this->_executionLogFile;
+		$log = null;
 
 		if (!$message) return;
 
 		if ($type) {
-			$log[] = '[' . Core::getCurrentDate() . '] ' . '[' . __($type) . '] ' . $message;
+			$log = '[' . Core::getCurrentDate() . '] ' . '[' . __($type) . '] ' . $message;
 		} else {
-			$log[] = $message;
+			$log = $message;
 		}
 
-		$this->_executionLog = $log;
+		$fp = fopen($logFile, 'ab');
+		if (flock($fp, LOCK_EX)) {
+			fwrite($fp, $log . PHP_EOL);
+			flock($fp, LOCK_UN);
+		} else {
+			// Couldn't lock the file.
+			assert(false);
```

Comment:
I prefer to use fatalError in cases like this. If assertions are disabled on the server, it'll break without warning anyone if the lock can't be made. It's open to debate, though.